src/NHibernate/Hql/Ast/ANTLR/Tree/IdentNode.cs

```diff
@@ -39,7 +39,7 @@ public override IType DataType
 					return fe.DataType;
 				}
 				ISQLFunction sf = Walker.SessionFactoryHelper.FindSQLFunction(Text);
-				return sf == null ? null : sf.ReturnType(null, null);
+				return sf?.ReturnType(null, Walker.SessionFactoryHelper.Factory);
```

Comment:
Unique case I have spotted where a methods chain of calls ending by calling `IType.SqlTypes` was not supplying an `IMapping` (the factory in this case).